src/libsyntax/parse/parser.rs

```diff
@@ -2936,6 +2936,18 @@ impl<'a> Parser<'a> {
                 self.err_dotdotdot_syntax(self.token.span);
             }
 
+            if self.token == token::LArrow {
```

Comment:
To avoid cluttering this method, I would prefer to put this in its own method:
```rust
self.maybe_ban_placement_suggest_comparison();
```

Reply:
I moved the error portion to a function called `err_larrow_operator` analogous to `err_dotdotdot_syntax`. I hope that's ok.

Reply:
Sure :)